src/Api/DataTransformers/JobInputDataTransformer.php

```diff
@@ -100,17 +100,22 @@ private function setPostScripts (Job $job, $postScripts)
                 $job->removePostScript($script);
             }
         }
-        $repository = $this->entityManager->getRepository('App:Script');
-        foreach ($postScripts as $script) {
-            $result = $repository->find($script);
-            if (null != $result) {
-                if ($result->getIsJobPost()) {
-                    $job->addPostScript($result);
-                }else {
-                    throw new InvalidArgumentException(sprintf('Script "%s" is not a job post script', $result->getId()));
+        if (null != $postScripts){
+            $repository = $this->entityManager->getRepository('App:Script');
+            $query = $repository->createQueryBuilder('s');
+            foreach ($postScripts as $script) {
+                $query = $repository->createQueryBuilder('s');
```

Comment:
Seems we can omit one of the two createQueryBuilder calls? The same applies for setPreScripts too